### PR TITLE
fixed not updating cursor on osx

### PR DESCRIPTION
**Pull request by boombuler**
the cursor on osx was only updated after the mouse was moved.

### CODE REVIEW

**Comment by boombuler** [comment]
Hi,
this might be a problem if the cursor is not over the current view. I'll have a look at it

File: src/OSX/Avalonia.MonoMac/TopLevelImpl.cs [hunk 1]
```diff
@@ -299,10 +304,17 @@ public override void ScrollWheel(NSEvent theEvent)
...
             public override void MouseExited(NSEvent theEvent)
             {
+                _isMouseOver = false;
                 MouseEvent(theEvent, RawMouseEventType.LeaveWindow);
                 base.MouseExited(theEvent);
             }
 
+            public override void MouseEntered(NSEvent theEvent)
+            {
+                _isMouseOver = true;
```
Comment (kekekeks) [inline src/OSX/Avalonia.MonoMac/TopLevelImpl.cs]: We might want to call _cursor.Set here

Comment (kekekeks) [reply src/OSX/Avalonia.MonoMac/TopLevelImpl.cs]: Or even call UpdateCursor

Comment (boombuler) [reply src/OSX/Avalonia.MonoMac/TopLevelImpl.cs]: That should not be necessary. If I understand the documentation of NSView correctly, `AddCursorRect` should update the cursor once the mouse enters the given frame.
The problem was if the cursor was already within the frame, the cursor was not updated. Thats why i added the `_cursor.Set()`

**Pull request merged**